avaframe/com1DFA/deriveParameterSet.py

```diff
@@ -197,3 +194,254 @@ def checkRelEntThVariation(cfg, variationDict):
+    idList = inputSimFiles[fName]['id']
+
+    # create key names for flags and parameter variation info
+    thFlag = thType + 'FromShp'
+    thPercent = thType + 'PercentVariation'
+
+    # if thickness should be read from shape file
+    if cfg['GENERAL'].getboolean(thFlag):
+        # if at least one but not all features in a shapefile have a thickness value - error
+        if ('None' in thicknessList):
+            message = 'Not all features in shape file have a thickness value - check shape file attributes: %s' % fName
+            log.error(message)
+            raise AssertionError(message)
+        else:
+            # set thickness value in ini file from info of shape file
+            thId = idList[0]
+            thThickness = thicknessList[0]
+            for count, id in enumerate(idList[1:]):
+                thId = thId +'|' + id
+                thThickness = thThickness + '|'+ thicknessList[count+1]
+
+            # add in INPUT section
+            cfg['INPUT'][thType + 'Id'] = thId
+            cfg['INPUT'][thType + 'Thickness'] = thThickness
+
+    else:
+        # if thickness should be read from ini file - check if format is correct
+        if '$' in cfg['GENERAL'][thType] and len(cfg['GENERAL'][thType].split('$')) != 3:
+            message = 'Format of relTh value in ini file is not correct - for variation from ini use refValue$percent$numberOfSteps'
+            log.error(message)
+            raise AssertionError(message)
+
+    return cfg
+
+
+def checkThicknessSettings(cfg, thName):
+    """ check if thickness setting format is correct
+
+        Parameters
+        ------------
+        cfg: configparser
+            configuration settings
+        thName: str
+            thickness parameter name (entTh, ...)
+
+        Returns
+        -------
+        thicknessSettingsCorrect: bool
+            True if settings are provided in the correct format
+
+    """
+
+    # create key name for thickness flag
+    thFlag = thName + 'FromShp'
+
+    # check if flag is set correctly and thickness parameter has correct format
+    if cfg['GENERAL'][thFlag] == 'True' or cfg['GENERAL'][thFlag] == 'False':
+        if cfg['GENERAL'].getboolean(thFlag):
+            if cfg['GENERAL'][thName] != '':
+                message = 'If %s is set to True - it is not allowed to set a value for %s' % (thFlag, thName)
+                log.error(message)
+                raise AssertionError(message)
+        else:
+            if cfg['GENERAL'][thName] == '':
+                message = 'If %s is set to False - it is required to set a value for %s' % (thFlag, thName)
+                log.error(message)
+                raise AssertionError(message)
+
+    else:
+        message = 'Check %s - needs to be True or False' % (thFlag)
+        log.error(message)
+        raise AssertionError(message)
+
+    # if no error ocurred - thickness settings are correct
+    thicknessSettingsCorrect = True
+
+    return thicknessSettingsCorrect
```

Comment:
why do you return something? it fails anyways if there is a problem

Reply:
I think it makes sense to return True in case you want to use it in the calling function. Otherwise the return value is None... (https://realpython.com/python-return-statement/#implicit-return-statements)

Reply:
ok but then do we need to catch the value when we call the function?

Reply:
no, not necessary. But possible if needed :-)

Reply:
so I leave it like it is now?